src/chains/filecoin/filecoin/src/things/address.ts

```diff
@@ -296,8 +311,90 @@ class Address extends SerializableLiteral<AddressConfig> {
       }`
     );
   }
+
+  static createChecksum(protocol: AddressProtocol, payload: Buffer): Buffer {
+    // Create a checksum using the blake2b algorithm
+    const checksumBuffer = Buffer.concat([Buffer.from([protocol]), payload]);
+    const checksum = blake.blake2b(
+      checksumBuffer,
+      null,
+      Address.CHECKSUM_BYTES
+    );
+    return Buffer.from(checksum.buffer);
+  }
+
+  static validate(inputAddress: string): Address {
+    inputAddress = inputAddress.trim();
+
+    if (inputAddress === "" || inputAddress === "<empty>") {
+      throw new Error("invalid address length");
+    }
+
+    // MaxAddressStringLength is the max length of an address encoded as a string
+    // it includes the network prefix, protocol, and bls publickey (bls is the longest)
+    const MaxAddressStringLength = 2 + 84;
+    if (
+      inputAddress.length > MaxAddressStringLength ||
+      inputAddress.length < 3
+    ) {
+      throw new Error("invalid address length");
+    }
+
+    const address = new Address(inputAddress);
+    const raw = address.value.slice(2);
+
+    if (address.network === AddressNetwork.Unknown) {
+      throw new Error("unknown address network");
+    }
+
+    if (address.protocol === AddressProtocol.Unknown) {
+      throw new Error("unknown address protocol");
+    }
+
+    if (address.protocol === AddressProtocol.ID) {
+      if (raw.length > 20) {
+        throw new Error("invalid address length");
+      }
+      const id = parseInt(raw, 10);
+      if (isNaN(id) || id.toString(10) !== raw) {
+        throw new Error("invalid address payload");
+      }
+      return address;
+    }
+
+    const customBase32Alphabet = "abcdefghijklmnopqrstuvwxyz234567";
```

Comment:
I know `customBase32Alphabet` is used in at least 1 other place. Could you make it a constant, or maybe just have a common custom base32.parse function?